### PR TITLE
feat(project): create interface and add schema validation to step 5,6

### PR DESCRIPTION
**Pull request by AzizSandid**
## What does this do?

create interface and add schema validation to step 5,6

### CODE REVIEW

**Review by MohamedMegdich1 — CHANGES_REQUESTED** [review]

File: src/modules/client/create-announcement/steps/choose-rooms-to-renovate.tsx [hunk 1]
```diff
@@ -1,28 +1,70 @@
...
   onHandleBack,
   onHandleNext,
+  setFormData,
+  formData,
 }: StepperFormProps) {
+  const { handleSubmit, control, errors } = useCustomForm(
+    CreateAnnouncementStepSixSchema,
+    { piecesRenovate: formData?.piecesRenovate }
+  );
+  console.log('error : ', errors);
```
[the remark this answers is not in the document]
Comment (AzizSandid) [reply src/modules/client/create-announcement/steps/choose-rooms-to-renovate.tsx]: fixed

File: src/modules/client/create-announcement/create-announcement-stepper.tsx [hunk 2]
```diff
@@ -22,7 +22,7 @@ const CreateAnnouncementInner = () => {
   const lastStep = 11;
   const { step, onHandleBack, onHandleNext, setFormData, formData } =
     useFormStepper<AnnouncementType>();
-
+  console.log('formData :', formData);
```
[the remark this answers is not in the document]
Comment (AzizSandid) [reply src/modules/client/create-announcement/create-announcement-stepper.tsx]: fixed

File: src/shared/components/counter.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  control,
+  rules,
+}: CounterProps<T>) => {
+  const { field } = useController({ control, name, rules });
+  const initialValue =
+    field.value?.find((item: any) => Object.keys(item)[0] === String(id))?.[
+      id
+    ] ?? 0;
+  const [count, setCount] = useState<number>(initialValue);
+  console.log('dd');
```
[the remark this answers is not in the document]
Comment (AzizSandid) [reply src/shared/components/counter.tsx]: fixed

File: src/modules/client/create-announcement/steps/choose-category.tsx [hunk 4]
```diff
@@ -10,6 +10,36 @@ import type { StepperFormProps } from '@/types';
 import type { AnnouncementType } from '@/types/announcement';
 
 import { CreateAnnouncementStepThreeSchema } from '../schemas';
+
+type CategoryData = {
```
[the remark this answers is not in the document]
Comment (AzizSandid) [reply src/modules/client/create-announcement/steps/choose-category.tsx:14]: fixed 
image

File: src/modules/client/create-announcement/steps/choose-rooms-to-renovate.tsx [hunk 5]
```diff
@@ -1,28 +1,72 @@
...
+import { Home } from '@/assets/icons';
+import { useCustomForm } from '@/core';
+import { StepButtons } from '@/modules/shared';
+import { Counter, View } from '@/shared/components';
 import type { StepperFormProps } from '@/types';
+import type { AnnouncementType } from '@/types/announcement';
+
+import { CreateAnnouncementStepSixSchema } from '../schemas';
+
+type RenovateTypeData = {
```
[the remark this answers is not in the document]
Comment (AzizSandid) [reply src/modules/client/create-announcement/steps/choose-rooms-to-renovate.tsx:13]: fixed

File: src/modules/client/create-announcement/steps/choose-property-type.tsx [hunk 6]
```diff
@@ -10,6 +10,19 @@ import type { StepperFormProps } from '@/types';
 import type { AnnouncementType } from '@/types/announcement';
 
 import { CreateAnnouncementStepFourSchema } from '../schemas';
+
```
[the remark this answers is not in the document]
Comment (AzizSandid) [reply src/modules/client/create-announcement/steps/choose-property-type.tsx:13]: fixed

File: src/modules/client/create-announcement/steps/choose-preferred-style.tsx [hunk 7]
```diff
@@ -8,6 +8,42 @@ import type { StepperFormProps } from '@/types';
 import type { AnnouncementType } from '@/types/announcement';
 
 import { CreateAnnouncementStepNineSchema } from '../schemas';
+
+type PreferredStyleData = {
```
[the remark this answers is not in the document]
Comment (AzizSandid) [reply src/modules/client/create-announcement/steps/choose-preferred-style.tsx:12]: fixed

File: src/modules/client/create-announcement/steps/choose-needs.tsx [hunk 8]
```diff
@@ -10,6 +10,52 @@ import type { StepperFormProps } from '@/types';
 import type { AnnouncementType } from '@/types/announcement';
 
 import { CreateAnnouncementStepTwoSchema } from '../schemas';
+
```
[the remark this answers is not in the document]
Comment (AzizSandid) [reply src/modules/client/create-announcement/steps/choose-needs.tsx:13]: fixed

File: src/modules/client/create-announcement/steps/choose-additional-info.tsx [hunk 9]
```diff
@@ -10,6 +10,29 @@ import type { StepperFormProps } from '@/types';
 import type { AnnouncementType } from '@/types/announcement';
 
 import { CreateAnnouncementStepTenSchema } from '../schemas';
+
+type ExtensionsData = {
```
[the remark this answers is not in the document]
Comment (AzizSandid) [reply src/modules/client/create-announcement/steps/choose-additional-info.tsx:14]: fixed

File: src/shared/components/counter.tsx [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+  rules?: RegisterOptions;
+};
+
+type PiecesRenovateType = { [key: number]: number };
+
+const findInitialValue = (values: PiecesRenovateType[], id: number): number => {
+  return values?.find((item) => Object.keys(item)[0] === String(id))?.[id] ?? 0;
+};
+
+const updateArray = (
```
[the remark this answers is not in the document]
Comment (AzizSandid) [reply src/shared/components/counter.tsx]: fixed

File: src/shared/hooks/use-counter-utils.tsx [hunk 11]
```diff
@@ -0,0 +1,30 @@
+type ItemType = { [key: number]: number };
```
Comment (MohamedMegdich1) [inline src/shared/hooks/use-counter-utils.tsx:1]: to review

File: src/shared/components/counter.tsx [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+  rules,
+}: CounterProps<T>) => {
+  const { findInitialValue, updateArray } = useCounterUtils();
+  const { field } = useController({ control, name, rules });
+  const initialValue = findInitialValue(field.value, id);
+  const [count, setCount] = useState<number>(initialValue);
+
+  const handleAdd = () => {
+    if (count < 50) {
+      const updatedValue = updateArray(field.value, id, count + 1);
```
Comment (MohamedMegdich1) [inline src/shared/components/counter.tsx:46]: remove count

**Review by MohamedMegdich1 — CHANGES_REQUESTED** [review]

**Review by MohamedMegdich1 — APPROVED** [review]

**Pull request merged**